index.js

```diff
@@ -326,8 +326,8 @@ function buildCode (location) {
   const propertiesLocation = mergeLocation(location, 'properties')
   Object.keys(schema.properties || {}).forEach((key) => {
     let propertyLocation = mergeLocation(propertiesLocation, key)
-    if (propertyLocation.$ref) {
-      propertyLocation = resolveRef(location, propertyLocation.$ref)
+    if (propertyLocation.schema.$ref) {
```

Comment:
was this till now dead code?

Reply:
This check exists here and in the buildValue code, they are duplicating each other. 
https://github.com/fastify/fast-json-stringify/blob/4a4251a2a37523bedc5a4962612534f8e03f9f90/index.js#L849.